### PR TITLE
VideoPress block: Fix Cancel button and provide better error message

### PR DESCRIPTION
**Pull request by leogermani**
This PR adds a better error message when there's an error not related to the upload itself.

Captura de tela de 2022-06-15 10-33-15

The image above shows the generic error message we see in all cases.

For example, when a user uploads a video in a format that is not supported the error message is not only incomplete but also misleading, as it recommends the user to check their connection. This error only happens when the video upload is successful, so there's nothing wrong with the connection or the upload process.

I believe the case above is pretty common.

Another situation is when a user of the Free plan tries to upload their second video. They will see this generic and misleading error message. This PR fixes it.

This PR also fixes a bug that was preventing the "Cancel" button of the error component from working.

#### Changes proposed in this Pull Request:

* Fix Cancel button on VideoPress block and provided better error message when video foramt is not supported

#### Jetpack product discussion

NA

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Activate VideoPress on a site running the Free plan.

Confirm the error before checking out this branch:

* Try to upload a unsupported video format (for example, .mkv).
* Confirm the error message says "An error was encountered during the upload. Check your network connection."
* Click the "Cancel" button and confirm it does not work and throws error in the console

Confirm the fix by checking out this branch and building Jetpack:

* Try to upload a unsupported video format (for example, .mkv).
* Confirm you see a specific error message about the format of the video, with a link to the support docs

Captura de tela de 2022-06-17 10-31-12

* Click Cancel and confirm it works, by resetting the Video block
* Now upload a valid video and confirm it gets added to the post
* Try to upload a second video and confirm you see an error message suggesting you to upgrade:

Captura de tela de 2022-06-15 15-04-05

### CODE REVIEW

**Comment by leogermani** [comment]
Asking a review from @roundhill as he implemented this dialog.

It took me a while to understand why I couldn't access the error from the API response so I just thought it would be better and easier to do it this way rather than dive to deep into the tus js client.

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js [hunk 1]
```diff
@@ -106,6 +106,39 @@ export default function ResumableUpload( { file } ) {
...
+							<a
+								href="https://wordpress.com/support/videopress/recommended-video-settings/"
+								target="_blank"
+								rel="noreferrer"
+							>
+								{ __( 'Check the recommended video settings.', 'jetpack' ) }
+							</a>
+						</>
+					);
+				} else {
```
Comment (roundhill) [inline projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:134]: Would it be possible to add another case here for the free video upload used string, and link it to the upgrade page? We could link the `Upgrade to a VideoPress plan` text like:

[Upgrade to a VideoPress plan](#) to unlock more videos and 1TB of storage.

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js [hunk 2]
```diff
@@ -106,6 +106,39 @@ export default function ResumableUpload( { file } ) {
 
 	const fileSizeLabel = filesize( file.size );
 
+	const getErrorMessage = () => {
+		let errorMessage = __(
+			'An error was encountered during the upload. Check your network connection.',
+			'jetpack'
+		);
+		if ( typeof error === 'object' ) {
+			const apiResponse = error.toString().match( /message":"([^"]+)"/ );
```
Comment (roundhill) [inline projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:115]: Is there something we could change on server side to prevent having to use this regex? We'd be happy to update it.

**Review by roundhill — DISMISSED** [review]
It tested well for me, it'd be cool if we could link to the upsell but not required 😄

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js [hunk 3]
```diff
@@ -106,6 +106,39 @@ export default function ResumableUpload( { file } ) {
...
+			if ( typeof apiResponse === 'object' && apiResponse.length === 2 ) {
+				const apiResponseMessage = apiResponse[ 1 ];
+				// Let's give this error a better message.
+				if ( apiResponseMessage === 'Invalid Mime' ) {
+					errorMessage = (
+						<>
+							{ __( 'The format of the video you uploaded is not supported.', 'jetpack' ) }
+							&nbsp;
+							<a
+								href="https://wordpress.com/support/videopress/recommended-video-settings/"
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:126]: Should we be using `getRedirectUrl()` here for that link?

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js [hunk 4]
```diff
@@ -106,6 +106,39 @@ export default function ResumableUpload( { file } ) {
...
+			// tus doesnt give us direct acces to the API response, but let's try to parse it to provide useful feedback for the user.
+			if ( typeof apiResponse === 'object' && apiResponse.length === 2 ) {
+				const apiResponseMessage = apiResponse[ 1 ];
+				// Let's give this error a better message.
+				if ( apiResponseMessage === 'Invalid Mime' ) {
+					errorMessage = (
+						<>
+							{ __( 'The format of the video you uploaded is not supported.', 'jetpack' ) }
+							&nbsp;
+							<a
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:125]: Since that link brings you outside of wp-admin, should we use `ExternalLink` here?

**Review by jeherve — DISMISSED** [review]

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 3 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:126]: See https://github.com/Automattic/jetpack/pull/23376

TL;DR is we are not doing it for blocks yet because we don't want simple site users to be redirected to `jetpack.com` and then back to `wordpress.com`.

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 4 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js]: Good point. Will have a look

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 2 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:115]: I'm not sure. This is something that happens in the tus-client lib and at some point I gave up looking for it in hopes that maybe you would have the answer :D Maybe it has to do with the response code the server sends

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 1 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:134]: Yeah, I thought about it but the PR was getting big enough. Also, I wanted to have some feedback in this regex approach before moving forward with more things

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 4 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js]: Updated it to use external link

Captura de tela de 2022-06-17 10-31-12

**Comment by robertbpugh** [comment]
The two error messages look good to me copy-wise.

**Review by samiff — APPROVED** [review]

**Comment by leogermani** [comment]
r247743-wpcom

File: projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js (hunk 2 above)
Comment (leogermani) [reply projects/plugins/jetpack/extensions/blocks/videopress/resumable-upload/index.js:115]: For future reference, the problems seems to be that the TUS protocol does not expect the response to be a json, and also the client does not handle errors on "creation" very well. See p1655822877174109-slack-CBG1CP4EN
